lib/ex_unit/lib/ex_unit/runner.ex

```diff
@@ -241,9 +259,19 @@ defmodule ExUnit.Runner do
       end
 
     EM.test_finished(config.manager, test)
+    handle_result(config, test)
     test
   end
 
+  defp handle_result(config, %{state: {:failed, _}}) do
+    failures = MaxFailureCounter.increase(config.fail_counter)
```

Comment:
Is there a way to not use an agent? We should avoid mutable state as much as possible. I would prefer to track this in the runner state itself.